src/Http/Cookie/CookieCollection.php

```diff
@@ -61,7 +62,24 @@ public static function createFromHeader(array $header)
     {
         $cookies = static::parseSetCookieHeader($header);
 
-        return new CookieCollection($cookies);
+        return new static($cookies);
+    }
+
+    /**
+     * Create a new collection from the cookies in a ServerRequest
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The array of set-cookie header values.
+     * @return static
+     */
+    public static function createFromServerRequest(ServerRequestInterface $request)
+    {
+        $data = $request->getCookieParams();
+        $cookies = [];
+        foreach ($data as $name => $value) {
+            $cookies[] = new Cookie($name, $value);
```

Comment:
Probably for when it merge with other cookie 
Because cookieCollection will use name;path;domain key
Even that method can be have second arg for include/not include path/domain ,that people able to choose one of that